src/main/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -587,41 +586,98 @@ private int unpackInt32(final byte[] buffer, final int offset) {
 
     public enum FileTermination {HAS_TERMINATOR_BLOCK, HAS_HEALTHY_LAST_BLOCK, DEFECTIVE}
 
+    /**
+     *
+     * @param file the file to check
+     * @return status of the last compressed block
+     * @throws IOException
+     */
     public static FileTermination checkTermination(final File file) throws IOException {
-        final long fileSize = file.length();
+        return checkTermination(file == null ? null : file.toPath());
```

Comment:
Won't this cause `checkTermination(Path)` to throw when `file == null`? If so avoiding the NPE here doesn't seem worthwhile. Maybe you want to return `FileTermination.DEFECTIVE` in that case?